### PR TITLE
delete copy/move ctor of MPMCQueue explicitly

### LINKED ISSUE #5329: Delete copy/move ctor of MPMCQueue explicitly
## Enhancement
MPMCQueue maybe used by different threads, copy and move it is dangerous.

### PR DESCRIPTION
**Pull request by guo-shaoge**
Signed-off-by: guo-shaoge <shaoge1994@163.com>

### What problem does this PR solve?

Issue Number: close #5329

Problem Summary: MPMCQueue maybe used by different threads, copy and move it is dangerous.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Common/MPMCQueue.h [hunk 1]
```diff
@@ -74,6 +74,13 @@ class MPMCQueue
             destruct(getObj(read_pos));
     }
 
+    // Cannot to use copy/move constructor,
+    // because MPMCQueue maybe used by different threads.
+    // Copy and move it is dangerous.
+    MPMCQueue(const MPMCQueue<T> &) = delete;
```
Comment (fuzhe1989) [inline dbms/src/Common/MPMCQueue.h:80]: please use `DISALLOW_COPY_AND_MOVE`.

Comment (guo-shaoge) [reply dbms/src/Common/MPMCQueue.h]: Done

**Review by fuzhe1989 — APPROVED** [review]

**Review by SeaRise — APPROVED** [review]

**Pull request merged**